src/app/login-page/login-page.component.html

```diff
@@ -0,0 +1,14 @@
+<div class="alert alert-danger" [hidden]="!error">
+  There was a problem logging in. Please try again.
+</div>
+<form role="form" (submit)="login()">
+  <div class="form-group">
+    <label for="username">Username:</label> <input type="text"
+                                                   class="form-control" id="username" name="username" [(ngModel)]="username"/>
```

Comment:
Try using a ReactiveForm if possible - While they are harder to implement at first - they are much extendable and easier to test in the long run.